src/main/drivers/system.c

```diff
@@ -146,14 +146,24 @@ uint32_t micros(void)
     return (ms * 1000) + (usTicks * 1000 - cycle_cnt) / usTicks;
 }
 
-inline uint32_t getCycleCounter(void)
+inline int32_t getCycleCounter(void)
 {
     return DWT->CYCCNT;
 }
 
-uint32_t clockCyclesToMicros(uint32_t clockCycles)
+inline int32_t clockCyclesToMicros(int32_t clockCycles)
```

Comment:
Why are we trying to force inlining here instead of trusting the compiler to inline where this matches the optimisation settings?

Reply:
This was just to keep these routines in keeping with the adjacent getCycleCounter() routine, but you make a good point, so I'll remove these. There are many examples of inline in the BF code. It's only a hint anyway, so perhaps we should remove all of these?

Reply:
Thanks. Yes, I think in most cases the compiler should be trusted to make the right calls about inlining based on optimisation levels.

Reply:
Can you clarify the position on using `inline` on functions in the BF developer documentation so that we all don't have to go down the 'to inline or not to inline' path?